### DISCUSSION TITLE
Using magic link and unable to grabe user info

### DISCUSSION THREAD

**Discussion by DomPatmore-SAE** [supabase/supabase#3704] [Questions]
HI all,
   I'm using magic link for authetication. I'm able to get the the JWT token on the callback, and I'm sending as a part of the fetch to an express server using the supabase client. However, when I try to setAuth() passing in the token, I only get a null user.

**Comment by silentworks** [comment]
Is the token definitely being set? also on the server, you should get the user via
```js
const { user } = await supabase.auth.api.getUserByCookie(request)
```
Also, make sure you are setting the auth cookie somewhere else in your code
```js
supabase.auth.api.setAuthCookie(request, response)
```

**Reply by silentworks** [reply]
Just pass the request to the `getUserByCookie` function above and it will get the token itself. Also please try and reply under the message here.

**Comment by DomPatmore-SAE** [comment]
Okay-I'm setting the token into the header of the request. Can I only pass
in the token via cookie?

**Comment by dompatmore** [comment]
I'm still unable to get any user info out. I'm adding more information for clarification.


I'm using a single HTML page with a express back-end (it's a project I've set up for students, no I'm not using any frameworks). I'm able to get a link to my inbox and clicking on it sends be back to the main page with the jwt token:
<img width="797" alt="Screenshot 2021-11-01 at 00 28 27" src="https://user-images.githubusercontent.com/36126777/139606705-ab99f2ea-101a-4733-b6d9-c52b6719a178.png">

What I do next is check if there is a hash in the location, grab that and send that to an auth endpoint on the backend
<img width="762" alt="Screenshot 2021-11-01 at 00 30 14" src="https://user-images.githubusercontent.com/36126777/139606782-ab7e61e9-6cd1-4901-90c7-7acb90ea4bf2.png">

As I'm not passing the details through a cookie, I'm not sure how I could that function; however, I tried to use this
<img width="738" alt="Screenshot 2021-11-01 at 00 32 42" src="https://user-images.githubusercontent.com/36126777/139606933-f3795fec-2a31-4eac-80c7-5ccfb2c741b8.png">

 and also <code>supabase.auth.getuser()</code> but each time the user is null.

Can I verify that a) this is the correct workflow for working with magic link, ie once I receive the token I should pass it along to my server and it should 'work'  b) Do I ever need to decode the token as I would with say Auth0?

I've tried looking around for examples, but the only ones I could see at the moment use NextJS.
